dask/dataframe/io/parquet/arrow.py

```diff
@@ -314,6 +354,99 @@ class ArrowDatasetEngine(Engine):
     # Public Class Methods
     #
 
+    @classmethod
+    def extract_filesystem(
+        cls,
+        urlpath,
+        dataset_options,
+        open_file_options,
+        storage_options,
+    ):
+
+        # Check if fs was already specified as a dataset option
+        filesystem = dataset_options.pop("filesystem", None)
+        if filesystem is None:
+            if isinstance(urlpath, (list, tuple, set)):
+                if not urlpath:
+                    raise ValueError("empty urlpath sequence")
+                strpath = stringify_path(next(iter(urlpath)))
+            else:
+                strpath = stringify_path(urlpath)
+            if (
+                strpath.startswith("s3://")
+                and not open_file_options
+                and not (
+                    # Only use PyArrow by default when storage is in s3,
+                    # and `storage_option` only includes simple options
+                    # # that are "expected" by `S3FileSystem`
+                    set(storage_options)
+                    - {
+                        "access_key",
+                        "secret_key",
+                        "session_token",
+                        "anonymous",
+                        "role_arn",
+                        "session_name",
+                        "external_id",
+                        "load_frequency",
+                        "region",
+                        "request_timeout",
+                        "connect_timeout",
+                        "scheme",
+                        "endpoint_override",
+                        "background_writes",
+                        "default_metadata",
+                        "proxy_options",
+                        "allow_bucket_creation",
+                        "allow_bucket_deletion",
+                        "retry_strategy",
+                    }
+                )
+            ):
+                filesystem = "arrow"
+            else:
+                filesystem = "fsspec"
+
+        if isinstance(filesystem, pa_fs.FileSystem) or filesystem == "arrow":
+            if isinstance(urlpath, (list, tuple, set)):
+                if not urlpath:
+                    raise ValueError("empty urlpath sequence")
+                urlpath = [stringify_path(u) for u in urlpath]
+            else:
+                urlpath = [stringify_path(urlpath)]
+
+            if filesystem == "arrow":
+                try:
+                    filesystem = type(pa_fs.FileSystem.from_uri(urlpath[0])[0])(
+                        **(storage_options or {})
+                    )
+                except (TypeError, pa.lib.ArrowInvalid) as err:
+                    # Try falling back to fsspec
+                    warnings.warn(
```

Comment:
This will result in a lot of warnings being emitted on existing code (e.g. any code that uses `storage_options` today). I'd prefer to not warn in this case and instead raise an informative error message if both `pyarrow` and the `s3fs` fallback behavior fail.

Reply:
Yeah, I changed my mind about this during our offline discussion. I’ll try to revise tomorrow AM